### PR TITLE
NNS1-2857: Replace icpAccountsStore with derived icpAccountsStore

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

The current icpAccountStore holds a combination of accounts and balances in a single store. This means all the data has to be loaded together and if we just want to change a balances, we have to manipulate the data in the store, which can interfere with loading the accounts.

The cleaner way is to have a separate store for accounts and a separate store for balances and a derived store which merged both into accounts with balances. That way when we update a balance in the balances store, it will just automatically recreate the accounts with balances.

The derived accounts store has already been created in previous PRs.

In this PR, we replace the old `icpAccountsStore` with the dervied `icpAccountsStore` for reading, and fill the `icpAccountDetailsStore` and `icpAccountBalanacesStore` for writing.

There are several flows in which the accounts store is filled. In order to keep all of them working the same, I tried to keep the same code structure of first querying both accounts and balances and then filling stores. If we did it from scratch, we'd probably separate these stops more but by keeping the same structure it's easier to guarantee that we keep the same behavior.

# Changes

1. Replace `loadAccounts` with a function `loadAccountsStoresData` that loads account details and balances separately and returns both.
2. Change the error handler to reset both the `icpAccountDetailsStore` and the `icpAccountBalancesStore`.
3. Change `loadBalance` to update the `icpAccountBalancesStore` instead of the old `icpAccountsStore`
4. Temporarily change `frontend/src/lib/stores/icp-accounts.store.ts` to just export the new derived store so we can change all the clients in a separate PR.

# Tests

1. Remove the tests for `loadAccounts` as it has been replaced by `loadAccountsData` which is not exported out of the file.
2. Add extra expectations to check the new stores in addition to the existing expectations which now check the derived store.
3. Remove `frontend/src/tests/lib/stores/icp-accounts.store.spec.ts`
4. Update `setAccountsForTesting`, `setAccountBalanceForTesting` and `resetAccountsForTesting` to work on the new stores.
5. Tested manually at https://qsgjb-riaaa-aaaaa-aaaga-cai.dskloet-ingress.devenv.dfinity.network/

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: frontend/src/lib/services/icp-accounts.services.ts [hunk 1]
```diff
@@ -108,55 +105,37 @@ export const loadAccounts = async ({
...
 
-  const [main, subAccounts, hardwareWallets] = await Promise.all([
-    mapAccount("main")(mainAccount),
-    Promise.all(mainAccount.sub_accounts.map(mapAccount("subAccount"))),
-    Promise.all(
-      mainAccount.hardware_wallet_accounts.map(mapAccount("hardwareWallet"))
-    ),
-  ]);
+  const balancesData = Object.fromEntries(
+    await Promise.all(
```
Comment (mstrasinskis) [inline frontend/src/lib/services/icp-accounts.services.ts:116]: Out of curiosity, is there a plan to use `Promise.allSettled` here?

File: frontend/src/lib/services/icp-accounts.services.ts [hunk 2]
```diff
@@ -513,29 +521,34 @@ const pollLoadAccounts = async (params: {
...
     });
-    mutableStore.set(certifiedAccounts);
+    icpAccountDetailsStore.set(accountDetailsData);
+    mutableBalancesStore.set({
+      data: balancesData,
+      certified: overrideCertified,
+    });
   } catch (err) {
-    mutableStore.cancel();
+    mutableBalancesStore.cancel();
```
Comment (mstrasinskis) [inline frontend/src/lib/services/icp-accounts.services.ts:551]: Is this enough w/o  `icpAccountDetailsStore` canceling?

**Review by mstrasinskis — APPROVED** [review]
Thanks, looks good! Just a couple of question.

File: frontend/src/lib/services/icp-accounts.services.ts (hunk 1 above)
Comment (dskloetd) [reply frontend/src/lib/services/icp-accounts.services.ts:116]: I'm not sure what you have in mind.
In the current flows we reset the stores if filling them fails.
Are you suggesting we completely changing the flows or something specific?

File: frontend/src/lib/services/icp-accounts.services.ts (hunk 2 above)
Comment (dskloetd) [reply frontend/src/lib/services/icp-accounts.services.ts:551]: canceling is specific to `queuedStore`.
`icpAccountDetailsStore` is not a `queuedStore` so there's nothing to cancel.

**Pull request merged**